music_assistant/providers/ibroadcast/__init__.py

```diff
@@ -372,14 +372,22 @@ async def _parse_track(self, track_obj: dict[str, Any]) -> Track:
                 )
             },
         )
+        if track_obj["album_id"]:
+            album = await self._client.get_album(track_obj["album_id"])
```

Comment:
Oof, this is heavy to do in the parse method of a track. Please make this optional because you're spamming api calls now.

Reply:
Yes, I know, but I need that anyway to set the album info a few lines below (431), or how is that supposed to happen?

Reply:
Only in the "get_track" method you should return a full track including album.
In regular (library) listings etc. you can simply return an ItemMapping as album (which is basically just a reference).

So. do you have something like the album id and the album name so you can creat ethat itemmapping ?

Otherwise imagine a 2000 tracks listing, you wil be doing 2000 individual calls to get the album, which is inefficient.

Reply:
I have an album_id in the track info but that's it. I can enrich this in the library already but that only shifts the problem.
As you're aware, with the current ibroadcast API the library is already in memory, but that's not really an excuse. Still hoping for a better API in the near future.

Reply:
hmm ok, lets see how it goes then. Worst case you need to cache the album info a bit.

Reply:
thanks, just let me know, happy to help